warning: loop variable ‘kv’ of `some` type binds to a temporary constructed from type `some`

```shell
/home/USER/Development/OpenMs/src/openms/source/ANALYSIS/OPENSWATH/MRMFeatureFilter.cpp: In member function ‘void OpenMS::MRMFeatureFilter::FilterFeatureMap(OpenMS::FeatureMap&, const OpenMS::MRMFeatureQC&, const OpenMS::TargetedExperiment&)’:
/home/USER/Development/OpenMs/src/openms/source/ANALYSIS/OPENSWATH/MRMFeatureFilter.cpp:205:70: warning: loop variable ‘kv’ of type ‘const std::pair<OpenMS::String, std::pair<double, double> >&’ binds to a temporary constructed from type ‘const std::pair<const OpenMS::String, std::pair<double, double> >’ [-Wrange-loop-construct]
  205 |             for (const std::pair<String, std::pair<double, double>>& kv : filter_criteria.component_group_qcs.at(cg_qc_it).meta_value_qc)
      |                                                                      ^~
/home/USER/Development/OpenMs/src/openms/source/ANALYSIS/OPENSWATH/MRMFeatureFilter.cpp:205:70: note: use non-reference type ‘const std::pair<OpenMS::String, std::pair<double, double> >’ to make the copy explicit or ‘const std::pair<const OpenMS::String, std::pair<double, double> >&’ to prevent copying
/home/USER/Development/OpenMs/src/openms/source/ANALYSIS/OPENSWATH/MRMFeatureFilter.cpp: In member function ‘void OpenMS::MRMFeatureFilter::FilterFeatureMapPercRSD(OpenMS::FeatureMap&, const OpenMS::MRMFeatureQC&, const OpenMS::MRMFeatureQC&)’:
/home/USER/Development/OpenMs/src/openms/source/ANALYSIS/OPENSWATH/MRMFeatureFilter.cpp:399:70: warning: loop variable ‘kv’ of type ‘const std::pair<OpenMS::String, std::pair<double, double> >&’ binds to a temporary constructed from type ‘const std::pair<const OpenMS::String, std::pair<double, double> >’ [-Wrange-loop-construct]
  399 |             for (const std::pair<String, std::pair<double, double>>& kv : filter_criteria.component_group_qcs.at(cg_qc_it).meta_value_qc)
      |                                                                      ^~
/home/USER/Development/OpenMs/src/openms/source/ANALYSIS/OPENSWATH/MRMFeatureFilter.cpp:399:70: note: use non-reference type ‘const std::pair<OpenMS::String, std::pair<double, double> >’ to make the copy explicit or ‘const std::pair<const OpenMS::String, std::pair<double, double> >&’ to prevent copying
```

--- a/src/openms/include/OpenMS/FILTERING/TRANSFORMERS/SpectraMerger.h
+++ b/src/openms/include/OpenMS/FILTERING/TRANSFORMERS/SpectraMerger.h
@@ -836,7 +836,7 @@ protected:
         // collect peaks from all spectra
         // loop over spectra in blocks
         std::vector<std::pair<double, double> > mz_intensity_all; // m/z positions and peak intensities from all spectra
-        for (const auto it2 : it->second)
+        for (const auto& it2 : it->second)
         {
           // loop over m/z positions
           for (typename MapType::SpectrumType::ConstIterator it_mz = exp[it2.first].begin(); it_mz < exp[it2.first].end(); ++it_mz)

--- a/src/openms/source/ANALYSIS/OPENSWATH/MRMFeatureFilter.cpp
+++ b/src/openms/source/ANALYSIS/OPENSWATH/MRMFeatureFilter.cpp
@@ -202,7 +202,7 @@ namespace OpenMS
               }
             }
 
-            for (const std::pair<String, std::pair<double, double>>& kv : filter_criteria.component_group_qcs.at(cg_qc_it).meta_value_qc)
+            for (const std::pair<const String, std::pair<double, double>>& kv : filter_criteria.component_group_qcs.at(cg_qc_it).meta_value_qc)
             {
               bool metavalue_exists{ false };
               if (!checkMetaValue(features.at(feature_it), kv.first, kv.second.first, kv.second.second, metavalue_exists))
@@ -396,7 +396,7 @@ namespace OpenMS
               ++cg_tests_count;
             }
 
-            for (const std::pair<String, std::pair<double, double>>& kv : filter_criteria.component_group_qcs.at(cg_qc_it).meta_value_qc)
+            for (const std::pair<const String, std::pair<double, double>>& kv : filter_criteria.component_group_qcs.at(cg_qc_it).meta_value_qc)
             {
               if (!checkRange(filter_values.component_group_qcs.at(cg_qc_it).meta_value_qc.at(kv.first).second,
                 kv.second.first,

